honeybee_radiance/cli/multiphase.py

```diff
@@ -947,17 +952,29 @@ def add_aperture_group_blinds_command(
     try:
         model: Model = Model.from_file(model_file)
 
-        unique_aperture_groups = {}
-        # first gather all apertures and group by dynamic group identifier
-        for ap in model.apertures:
-            if isinstance(ap.boundary_condition, Outdoors):
-                dgi = ap.properties.radiance.dynamic_group_identifier
-                if dgi is not None:
-                    ap.properties.radiance.remove_states()
-                    if dgi in unique_aperture_groups:
-                        unique_aperture_groups[dgi].append(ap)
-                    else:
-                        unique_aperture_groups[dgi] = [ap]
+        def get_unique_aperture_groups(model):
+            unique_aperture_groups = {}
+            for ap in model.apertures:
+                if isinstance(ap.boundary_condition, Outdoors):
+                    dgi = ap.properties.radiance.dynamic_group_identifier
+                    if dgi is not None:
+                        ap.properties.radiance.remove_states()
+                        if dgi in unique_aperture_groups:
+                            unique_aperture_groups[dgi].append(ap)
+                        else:
+                            unique_aperture_groups[dgi] = [ap]
+
+            return unique_aperture_groups
+
+        unique_aperture_groups = get_unique_aperture_groups(model)
+
+        if not unique_aperture_groups and create_groups:  # no aperture groups, create them
+            model = automatic_aperture_grouping(
+                model, room_based=True, view_factor_or_orientation=False)
+        else:
+            raise ValueError('No aperture groups found in the model.')
+
+        unique_aperture_groups = get_unique_aperture_groups(model)
```

Comment:
Shouldn't we let the model that has aperture groups go through? Also I might be wrong but why would you need to recalculate the aperture groups twice?

```suggestion
        unique_aperture_groups = get_unique_aperture_groups(model)

        if unique_aperture_groups:
            # there are aperture groups in the model already
            pass
        elif not unique_aperture_groups and create_groups:  # no aperture groups, create them
            model = automatic_aperture_grouping(
                model, room_based=True, view_factor_or_orientation=False)
            unique_aperture_groups = get_unique_aperture_groups(model)
        else:
            raise ValueError(
                'No aperture groups found in the model. Either provide a model with aperture groups or '
                'use the --create-groups option to calculate the aperture groups as part of this command.'
            )

        
```

Reply:
You are right. I think I messed it up because I decided to change the code just before sending the PR. Your suggestion looks good to me.